airflow/www/views.py

```diff
@@ -1349,21 +1357,26 @@ class GraphForm(DateTimeWithNumRunsWithDagRunsForm):
         form = GraphForm(data=dt_nr_dr_data)
         form.execution_date.choices = dt_nr_dr_data['dr_choices']
 
-        task_instances = {
+        task_instances_dict = {
             ti.task_id: alchemy_to_dict(ti)
-            for ti in dag.get_task_instances(session, dttm, dttm)}
+            for ti in task_instances}
         tasks = {
             t.task_id: {
                 'dag_id': t.dag_id,
-                'task_type': t.task_type,
+                'task_type': t.operator,
             }
-            for t in dag.tasks}
+            for t in task_instances}
         if not tasks:
             flash("No tasks found", "error")
         session.commit()
         doc_md = markdown.markdown(dag.doc_md) \
             if hasattr(dag, 'doc_md') and dag.doc_md else ''
 
+        unique_ops = {
```

Comment:
Since you take only the values of this result, it would be shorter to write:

```
unique_ops = {{"name": op.operator, "ui_color": op.ui_color, "ui_fgcolor": op.ui_fgcolor} for op in task_instances}
```
So you don't have to construct a dict first.

Reply:
I have tried this but someone this does not work and the current code does work ;)

Reply:
Odd, the values are simply strings and I cannot think of anything going wrong here. Will check.